test/test/otlp/OtlpTests.java

```diff
@@ -42,19 +40,36 @@ public void testSampleType(TestProcess p) throws Exception {
         assert sampleType1.getAggregationTemporality() == AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE;
     }
 
+    @Test(mainClass = CpuBurner.class, agentArgs = "start,otlp,threads,file=%f.pb")
+    public void threadName(TestProcess p) throws Exception {
+        ProfilesData profilesData = waitAndGetProfilesData(p);
+
+        Profile profile = getFirstProfile(profilesData);
+        assert profile.getSampleTypeList().size() == 2;
+
+        Set<String> threadNames = new HashSet<>();
+        for (Sample sample: profile.getSampleList()) {
+            Optional<AnyValue> threadName = getAttribute(sample, profilesData.getDictionary(), "thread.name");
+            if (!threadName.isPresent()) continue;
+            threadNames.add(threadName.get().getStringValue());
+        }
+        boolean cpuBurnerFound = threadNames.stream().anyMatch("CpuBurnerWorker"::equals);
```

Comment:
Must-read post: https://medium.com/@tagir_valeev/use-stream-api-simpler-or-dont-use-it-at-all-ea0a44a4b1ff
Notably §1.6